pubnub-kotlin/pubnub-kotlin-impl/src/integrationTest/kotlin/com/pubnub/api/integration/SubscribeIntegrationTests.kt

```diff
@@ -1365,6 +1365,138 @@ class SubscribeIntegrationTests : BaseIntegrationTest() {
+            AtomicInteger(0) // status event emitted but no actual resubscribe when channels unchanged
+        val subscribeHttpRequestCount = AtomicInteger(0) // Count actual HTTP subscribe requests
+
+        // Track messages received by each listener
+        val listenerAMessageCount = AtomicInteger(0)
+        val listenerBMessageCount = AtomicInteger(0)
+
+        // Custom logger to count HTTP subscribe requests
+        val customLogger = object : CustomLogger {
+            override fun debug(logMessage: LogMessage) {
+                if (logMessage.type == LogMessageType.NETWORK_REQUEST) {
+                    val networkRequestDetails = logMessage.message as LogMessageContent.NetworkRequest
+                    if (networkRequestDetails.path.contains("/v2/subscribe/")) {
+                        subscribeHttpRequestCount.incrementAndGet()
+                        println("HTTP Subscribe request #${subscribeHttpRequestCount.get()}: ${networkRequestDetails.path}")
+                    }
+                }
+            }
+        }
+
+        clientConfig = {
+            customLoggers = listOf(customLogger)
+        }
+
+        pubnub.addListener(
+            object : StatusListener {
+                override fun status(
+                    pubnub: PubNub,
+                    status: PNStatus,
+                ) {
+                    when (status.category) {
+                        PNStatusCategory.PNConnectedCategory -> {
+                            connectedEventCount.incrementAndGet()
+                            connectedLatch.countDown()
+                        }
+
+                        PNStatusCategory.PNSubscriptionChanged -> {
+                            subscriptionChangedCount.incrementAndGet()
+                            subscriptionChangedLatch.countDown()
+                        }
+
+                        else -> {}
+                    }
+                }
+            }
+        )
+
+        // First subscription
+        val subscriptionSet1 = pubnub.subscriptionSetOf(setOf(channelName, channelName02))
+        subscriptionSet1.addListener(
+            object : EventListener {
+                override fun message(
+                    pubnub: PubNub,
+                    result: PNMessageResult,
+                ) {
+                    println("ListenerA received: ${result.message.asString}")
+                    listenerAMessageCount.incrementAndGet()
+                    messagesLatch.countDown()
+                }
+            }
+        )
+
+        // Second subscription (SAME channel, different listener)
+        val subscriptionSet2 = pubnub.subscriptionSetOf(setOf(channelName))
+        subscriptionSet2.addListener(
+            object : EventListener {
+                override fun message(
+                    pubnub: PubNub,
+                    result: PNMessageResult,
+                ) {
+                    println("ListenerB received: ${result.message.asString}")
```

Comment:
The same as [above](https://github.com/pubnub/kotlin/pull/384/files#r2545047277) goes here